### ISSUE TITLE
Use of wrong compiler flag attempted

### ISSUE THREAD

**Issue by jengelh** [semigroups/Semigroups#894]
```
m4/ax_check_hpcombi.m4:  AX_CHECK_COMPILE_FLAG(-march=avx,
libsemigroups/m4/ax_check_hpcombi.m4:    CXXFLAGS="-march=avx -flax-vector-conversions"
```

avx is not a CPU type; you probably wanted `-mavx` instead. This affects both `libsemigroups` and `semigroups`.

**Comment by jengelh** [comment]
Works.

**Comment by james-d-mitchell** [comment]
Resolved in v5.2.0
